### DISCUSSION TITLE
Drag multiple ListBox items

### DISCUSSION THREAD

**Discussion by IncPlusPlus** [AvaloniaUI/Avalonia.Xaml.Behaviors#109] [Q&A]
Hi there. Thanks for making such an awesome library. I was searching around trying to find out how to make an item draggable in Avalonia and almost gave up until I found this. I've figured out how to make a single `ListBox` item draggable. However, my `SelectionMode` is set to multiple so multiple items can be selected. When I select multiple items and then start to drag, all items except the one I started the drag action on get deselected. Is there a way to make dragging respect the existing selection of items in a `ListBox`?

I tried removing the drag & drop behavior and discovered that this issue seems to be with Avalonia specifically. If I start making a dragging motion, the selection is cleared.

**Comment by Tenjim** [comment]
Hi,
I have got the same issue with multiple selection. 
Is a new behavior based on a SelectionModel (SelectionModel from ListBox) can resolve this issue or it's very specific to Avalonia ?

**Reply by Tenjim** [reply]
Hi guys,

Do you know where is the behavior when we are selecting one ListBoxItem ? I'm looking to find a way to keep selection from listbox when begin Drag and Drop behavior.
I'm trying to create EventHandler on pointerpressed but after my event. Something select my ListBoxItem anyway.

thanks in advance.
